.github/workflows/publish_pypi.yml

```diff
@@ -1,38 +1,62 @@
 name: Build and publish package to PyPI
 
 on:
-  push:
-    tags:
-    - '*'
+  release:
+    types: [published]
   workflow_dispatch:
     inputs:
       target:
-        description: 'Deployment target. Can be "pypi" or "testpypi"'
-        default: "pypi"
+        description: 'Deployment target. Can be "pypi" or "testpypi", or blank to skip publishing.'
+        default: ""
 
 jobs:
-  publish_pypi:
-    name: Build wheels on ubuntu-latest
+  build:
+    name: Build sdist and wheel
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
-        with:
-          python-version: 3.8
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install build
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
       - name: Build package
-        run: python -m build
-      - name: Publish on PyPI
-        if: github.event.inputs.target == 'pypi'
+        run: pipx run build
+
+      - name: Upload sdist and wheel artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: distributions
+          path: dist/*
+
+  publish:
+    name: Publish to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
```

Comment:
This environment would be needed to set up in the PyPI settings before we are able to publish wheels.

Reply:
Also, it would be nice to have more than one reviewer that approves the deployment – the idea is that it should be very hard for someone unauthorised to push to PyPI.